h2o-algos/src/main/java/hex/glm/ConstrainedGLMUtils.java

```diff
@@ -0,0 +1,858 @@
+                                       DataInfo dinfo, List<String> coefNames, List<String> colNames) {
+    List<Integer> processedRowIndices = new ArrayList<>();
+    int constraintNumberFrame = (int) constraintF.vec("constraint_numbers").at(rowIndices.get(0));
+    LinearConstraints currentConstraint = new LinearConstraints();
+    String constraintType = constraintF.vec("types").stringAt(rowIndices.get(0)).toLowerCase();
+    boolean standardize = dinfo._normMul != null;
+    boolean constantFound = false;
+    for (Integer rowIndex : rowIndices) {
+      String coefName = constraintF.vec("names").stringAt(rowIndex);
+      String currType = constraintF.vec("types").stringAt(rowIndex).toLowerCase();
+      if (!coefNames.contains(coefName) && !"constant".equals(coefName))
+        throw new IllegalArgumentException("Coefficient name " + coefName + " is not a valid coefficient name.  It " +
+                "be a valid coefficient name or it can be constant");
+      if ((int) constraintF.vec("constraint_numbers").at(rowIndex) == constraintNumberFrame) {
+        if (!constraintType.equals(currType))
+          throw new IllegalArgumentException("Constraint type "+" of the same constraint must be the same but is not." +
+                  "  Expected type: "+constraintType+".  Actual type: "+currType);
+        if ("constant".equals(coefName))
+          constantFound = true;
+        processedRowIndices.add(rowIndex);
+        // coefNames is valid
+        if (standardize && colNames.contains(coefName)) {  // numerical column with standardization
+          int colInd = colNames.indexOf(coefName);
+          currentConstraint._constraints.put(coefName, constraintF.vec("values").at(rowIndex)*dinfo._normMul[colInd-dinfo._cats]);
+        } else {  // categorical column, constant or numerical column without standardization
+          currentConstraint._constraints.put(coefName, constraintF.vec("values").at(rowIndex));
+        }
+      }
+    }
+    if (!constantFound)
+      currentConstraint._constraints.put("constant", 0.0);  // put constant of 0.0
+    if (currentConstraint._constraints.size() < 3)
+      throw new IllegalArgumentException("Linear constraint must have at least two coefficients.  For constraints on" +
+              " just one coefficient: "+ constraintF.vec("names").stringAt(0)+", use betaConstraints instead.");
+    equalC.add(currentConstraint);
+    rowIndices.removeAll(processedRowIndices);
+  }
+  
+  public static double[][] formConstraintMatrix(ComputationState state, List<String> constraintNamesList, int[] betaEqualLessThanInd) {
+    // extract coefficient names from constraints
+    constraintNamesList.addAll(extractConstraintCoeffs(state));
+    // form double matrix
+    int numRow = (betaEqualLessThanInd == null ? 0 : ArrayUtils.sum(betaEqualLessThanInd)) +
+            (state._equalityConstraintsLinear == null ? 0 : state._equalityConstraintsLinear.length) + 
```

Comment:
@wendycwong, I tried to investigate, why the tests are failing and I found you probably don't calculate `numRow` correctly. 

```suggestion
           (state._equalityConstraintsBeta == null? 0 : state._equalityConstraintsBeta.length) +
            (state._equalityConstraintsLinear == null ? 0 : state._equalityConstraintsLinear.length) + 
```

I see in  `fillConstraintValues` method, you are doing:

```
if (state._equalityConstraintsBeta != null)
      rowIndex = extractConstraintValues(state._equalityConstraintsBeta, constraintNamesList, initCMatrix, rowIndex, null);
```

This change the `numRow` to be higher than `initCMatrix` size. If I try to fix this, the test is still failing, but on assert. So there is bug somewhere else yet. :)

Reply:
Veronika:

You are not kidding when you said you will do a detail review for me. You acquire so much knowledge to fix my errors!  Thank you.  I know who to ask to fix errors in constrained GLM in the future!!!!

Yes, your fix will work.  However, my intention is to use method extractBetaConstraints to count the number of beta constraints (equality or less than and equal to).

Note that the beta constraint 0.1 <= beta <= 0.2 will be extracted into two constraints but it is counted as once.  One equality constraint is counted as 1 which I failed to implement.  I fixed that.

Yes, there are other issues that I fixed for the test as well.